src/sentry/models/integration.py

```diff
@@ -147,3 +147,16 @@ def add_organization(self, organization, user=None, default_auth_id=None):
             )
 
             return org_integration
+
+    def reauthorize(self, data):
+        """
+        The structure of `data` depends on the `build_integration`
+        method on the integration provider.
+
+        Each provider may have their own way of reauthorizing the
+        integration.
+        """
+        if self.provider == "slack":
```

Comment:
@MeredithAnya I think we should throw an `IntegrationError` for any other provider since we haven't actually implemented this for anything except Slack.